src/packages/rte/tiptap/extensions/umb/mediapicker.extension.ts

```diff
@@ -1,63 +1,143 @@
+		const currentTarget = editor.getAttributes('image');
+		const figure = editor.getAttributes('figure');
+
+		let currentMediaUdi: string | undefined = undefined;
+		if (currentTarget?.['data-udi']) {
+			currentMediaUdi = getGuidFromUdi(currentTarget['data-udi']);
+		}
+
+		let currentAltText: string | undefined = undefined;
+		if (currentTarget?.alt) {
+			currentAltText = currentTarget.alt;
+		}
+
+		let currentCaption: string | undefined = undefined;
+		if (figure?.figcaption) {
+			currentCaption = figure.figcaption;
+		}
 
-		const selection = await this.#openMediaPicker();
+		const selection = await this.#openMediaPicker(currentMediaUdi);
 		if (!selection || !selection.length) return;
 
-		editor?.chain().focus().insertContent(`<umb-media>${selection}</umb-media>`).run();
+		const mediaGuid = selection[0];
+		const media = await this.#showMediaCaptionAltText(mediaGuid, currentAltText, currentCaption);
+		if (!media) return;
+
+		this.#insertInEditor(editor, mediaGuid, media);
 	}
 
-	async #openMediaPicker() {
-		const modalManager = await this.getContext(UMB_MODAL_MANAGER_CONTEXT);
-		const modalHandler = modalManager?.open(this, UMB_MEDIA_PICKER_MODAL, {
-			data: { multiple: false },
-			value: { selection: [] },
+	async #openMediaPicker(currentMediaUdi?: string) {
+		const modalHandler = this.#modalManager?.open(this, UMB_MEDIA_PICKER_MODAL, {
+			data: {
+				multiple: false,
+				//startNodeIsVirtual,
+			},
+			value: {
+				selection: currentMediaUdi ? [currentMediaUdi] : [],
+			},
 		});
 
 		if (!modalHandler) return;
 
 		const { selection } = await modalHandler.onSubmit().catch(() => ({ selection: undefined }));
 
-		//console.log('umb-media.selection', selection);
 		return selection;
 	}
+
+	async #showMediaCaptionAltText(mediaUnique: string, altText?: string, caption?: string) {
+		const modalHandler = this.#modalManager?.open(this, UMB_MEDIA_CAPTION_ALT_TEXT_MODAL, {
+			data: { mediaUnique },
+			value: {
+				url: '',
+				altText,
+				caption,
+			},
+		});
+		const mediaData = await modalHandler?.onSubmit().catch(() => null);
+		return mediaData;
+	}
+
+	async #insertInEditor(editor: Editor, mediaUnique: string, media: UmbMediaCaptionAltTextModalValue) {
+		if (!media?.url) return;
+
+		const { width, height } = await imageSize(media.url, { maxWidth: this.maxWidth });
+
+		const img = {
+			alt: media.altText,
+			src: media.url ? media.url : 'nothing.jpg',
+			'data-udi': `umb://media/${mediaUnique.replace(/-/g, '')}`,
+			width: width.toString(),
+			height: height.toString(),
+		};
+
+		if (media.caption) {
+			return editor.commands.insertContent({
+				type: 'figure',
+				content: [
+					{
+						type: 'paragraph',
+						content: [
+							{
+								type: 'image',
+								attrs: img,
+							},
+						],
+					},
+					{
+						type: 'figcaption',
+						content: [
+							{
+								type: 'text',
+								text: media.caption,
+							},
+						],
+					},
+				],
+			});
+		}
```

Comment:
This could be moved to a command, but what should own it, image or figure? Let's wait and see if other extensions could benefit from it down the road.